File: flatten_posts.py
import json
with open('/Users/farvascott/code/theethicsreporter/site/data/posts.json', 'r') as f:
    posts = json.load(f)

for p in posts:
    if isinstance(p.get('title'), dict) and 'rendered' in p['title']:
        p['title'] = p['title']['rendered']
    if isinstance(p.get('content'), dict) and 'rendered' in p['content']:
        p['content'] = p['content']['rendered']
    if isinstance(p.get('excerpt'), dict) and 'rendered' in p['excerpt']:
        p['excerpt'] = p['excerpt']['rendered']

with open('/Users/farvascott/code/theethicsreporter/site/data/posts.json', 'w') as f:
    json.dump(posts, f, indent=2)

print("Flattened posts.json")

File: tools/maintenance/flatten_posts.js
const fs = require('fs');
const dataPath = 'theethicsreporter.com/data/posts.json';
let data = JSON.parse(fs.readFileSync(dataPath, 'utf8'));

let modified = false;
for (let p of data) {
  if (p.title && typeof p.title === 'object' && p.title.rendered) {
    p.title = p.title.rendered;
    modified = true;
  }
  if (p.content && typeof p.content === 'object' && p.content.rendered) {
    p.content = p.content.rendered;
    modified = true;
  }
  if (p.excerpt && typeof p.excerpt === 'object' && p.excerpt.rendered) {
    p.excerpt = p.excerpt.rendered;
    modified = true;
  }
}

if (modified) {
  fs.writeFileSync(dataPath, JSON.stringify(data, null, 2));
  console.log("Flattened posts.json");
} else {
  console.log("No flattening needed");
}
